src/Symfony/Component/Console/Helper/ProgressBar.php

```diff
@@ -600,4 +614,23 @@ private static function initFormats()
             'debug_nomax' => ' %current% [%bar%] %elapsed:6s% %memory:6s%',
         );
     }
+
+    /**
+     * @param string $line
+     * @return bool
+     */
+    private function adjustBarWidthToWindowWidth($line)
+    {
+        $lineLength = Helper::strlenWithoutDecoration($this->output->getFormatter(), $line);
+        $windowWidth = $this->terminalDimensionsProvider->getTerminalWidth();
+        if ($lineLength > $windowWidth) {
+            $barWidthDelta = $lineLength - $windowWidth;
+            $newBarWidth = $this->barWidth - $barWidthDelta - 20;
+            $this->setBarWidth($newBarWidth);
```

Comment:
I've added check to the `setBarWidth()` method, since other methods depends on it.

It should never be negative.